sdk/pinocchio/src/account_info.rs

```diff
@@ -715,16 +725,26 @@ impl<'a, T: ?Sized> RefMut<'a, T> {
     pub fn map<U: ?Sized, F>(orig: RefMut<'a, T>, f: F) -> RefMut<'a, U>
     where
         F: FnOnce(&mut T) -> &mut U,
+    {
+        Self::try_map::<_, _, Infallible>(orig, |x| Ok(f(x))).unwrap()
+    }
+
+    /// Maps a mutable reference to a new type in a fallible map, returning an error if the map fails.
+    #[inline]
+    pub fn try_map<U: ?Sized, F, E>(orig: RefMut<'a, T>, f: F) -> Result<RefMut<'a, U>, E>
+    where
+        F: FnOnce(&mut T) -> Result<&mut U, E>,
     {
         // Avoid decrementing the borrow flag on Drop.
         let mut orig = ManuallyDrop::new(orig);
+        let value = f(&mut *orig)?;
```

Comment:
Can we add a test for the `try_map`? Mainly to test the behaviour of the drop in case of failure. I wonder whether we need to invert these lines or not, i.e., first call the function before wrapping the `orig` with a `ManuallyDrop`.